tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -0,0 +1,103 @@
+)
+from ocs_ci.framework.pytest_customization.marks import bugzilla
+
+
+logger = logging.getLogger(__name__)
+
+
+@tier4c
+@bugzilla("1992472")
+@pytest.mark.polarion_id("OCS-2705")
+class TestNonOCSTaintAndTolerations(E2ETest):
+    """
+    Test to test non ocs taints on ocs nodes
+    and toleration
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, nodes, request):
+        """
+        Make sure all nodes are up and untainted
+
+        """
+
+        def finalizer():
+
+            nodes.restart_nodes_by_stop_and_start_teardown()
+            assert untaint_nodes(taint_label="xyz=true:NoSchedule"), "Failed to untaint"
+
+        request.addfinalizer(finalizer)
+
+    def test_non_ocs_taint_and_tolerations(self, nodes):
+        """
+        1. Check if nodes are not ocs tainted
+        2. Taint ocs nodes with non-ocs taint
+        3. Set tolerations on storagecluster, subscription and configmap
+        4. Respin all ocs pods and check if it runs on ocs nodes with tolerations
+        5. Reboot worker nodes
+
+        """
+
+        # Check tolerations on pods under openshift-storage
+        assert not check_toleration_on_pods(), "Nodes already ocs tainted"
+
+        # Taint all nodes with non-ocs taint
+        ocp_nodes = get_all_nodes()
+        taint_nodes(nodes=nodes, taint_label="xyz=true:NoSchedule")
+
+        # Add tolerations to the storagecluster
+        storagecluster_obj = ocp.OCP(
+            resource_name="ocs-storagecluster",
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+            kind=constants.STORAGECLUSTER,
+        )
+        tolerations = (
+            '{"tolerations": [{"effect": "Noschedule", "key": "xyz",'
+            '"operator": "Equal", "value": "true"}, '
+            '{"effect": "Noschedule", "key": "node.ocs.openshift.io/storage", '
+            '"operator": "Equal", "value": "true"}]}'
+        )
+        param = (
+            f'{{"spec": {{"placement": {{"all": {tolerations}, "mds": {tolerations}, '
+            f'"noobaa-core": {tolerations}, "rgw": {tolerations}}}}}}}'
+        )
+        storagecluster_obj.patch(params=param, format_type="merge")
+
+        # Add tolerations to the subscription
+        sub_obj = ocp.OCP(
+            resource_name="odf-operator",
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+            kind=constants.SUBSCRIPTION,
+        )
+        param = (
+            '{"spec": {"config":  {"tolerations": '
+            '[{"effect": "Noschedule", "key": "xyz", "operator": "Equal", '
+            '"value": "true"}]}}}'
+        )
+        sub_obj.patch(params=param, format_type="merge")
+
+        # TODO Add tolerations to the configmap rook-ceph-operator-config
+
+        # Respin all pods and check it if is still running
+        pod_list = get_all_pods(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+        for pod in pod_list:
+            pod.delete(wait=False)
+        assert wait_for_pods_to_be_running(timeout=300)
```

Comment:
the BZ verification comment mentioned "Tool-box was in the Pending state because it did not have any tolerations, This will be tracked in a separate BZ". Is that BZ filed and fixed? If not. how is this handled here?

Reply:
bz https://bugzilla.redhat.com/show_bug.cgi?id=2012084

Reply:
without the fix for the bz, will there be assert as since toolbox pod will remain in pending state?

Reply:
I excluded toolbox pod from respins, but not sure how do we avoid it during node reboots. hence skipped the node reboots for now,Once the BZ is fixed we can change it

Reply:
ok. please create a github issue for node reboots to be added after BZ is fixed.